src/xscen/extract.py

```diff
@@ -1068,6 +1035,174 @@ def _get_warming_level(model):
+       Lists of dicts, strings or Datasets are also accepted, in which case the output will be a dict.
+       Regex wildcards (.*) are accepted, but may lead to unexpected results.
+       Datasets should include the catalogue attributes (starting by "cat:") required to create such a string:
+       'cat:mip_era', 'cat:experiment', 'cat:member',
+       and either 'cat:source' for global models or 'cat:driving_model' for regional models.
+       e.g. 'CMIP5_CanESM2_rcp85_r1i1p1'
+    period : sequence of str
+        [start, end] of the period to be evaluated.
+    tas_baseline_period : list, optional
+       [start, end] of the base period. The warming is calculated with respect to it. The default is ["1850", "1900"].
+    ignore_member : bool
+       Decides whether to ignore the member when searching for the model run in tas_csv.
+    tas_src : str, optional
+       Path to a netCDF of annual global mean temperature (tas) with an annual "time" dimension
+       and a "simulation" dimension with the following coordinates: "mip_era", "source", "experiment" and "member".
+       If None, it will default to data/IPCC_annual_global_tas.nc which was built from
+       the IPCC atlas data from  Iturbide et al., 2020 (https://doi.org/10.5194/essd-12-2959-2020)
+       and extra data for missing CMIP6 models and pilot models of CRCM5 and ClimEx.
+
+    Returns
+    -------
+    pd.Series or xr.DataArray or float
+        If `realization` is not a sequence, the output will be a float.
+        If `realization` is a sequence, the output will be of the same type.
+    """
+    tas_src = tas_src or Path(__file__).parent / "data" / "IPCC_annual_global_tas.nc"
+    tas_baseline_period = standardize_periods(
+        tas_baseline_period or ["1850", "1900"], multiple=False
+    )
+    period = standardize_periods(period, multiple=False)
+
+    FIELDS = ["mip_era", "source", "experiment", "member"]
+    info_models = _wl_prep_infomodels(realization, ignore_member, FIELDS)
+
+    # open nc
+    tas = xr.open_dataset(tas_src).tas
+
+    def _get_warming_level(model):
+        tas_sel = _wl_find_column(tas, model)
+        if tas_sel is None:
+            return None
+
+        selected = "_".join([tas_sel[c].item() for c in FIELDS])
+        msg = f"Computing warming level during {period} for {model} from simulation: {selected}."
+        logger.debug(msg)
+
+        if not all(
+            yr in tas_sel.time.dt.year
+            for yr in np.arange(int(period[0]), int(period[1]) + 1)
+        ):
+            raise ValueError(
+                f"Period {period} is not fully covered by the provided 'tas_src' database for {selected}."
+            )
+
+        # compute reference temperature for the warming and difference from reference
+        wl = (
+            tas_sel.sel(time=slice(*period)).mean()
+            - tas_sel.sel(time=slice(*tas_baseline_period)).mean()
+        )
```

Comment:
```suggestion
def IPCC_rolling(da, window=20, dim="time"): 
    out = da.rolling({dim:window}, center=True).mean()
    if window % 2 == 0:  # Even window
        out = out.shift({dim:-1})
    return out 
        wl = (
            IPCC_rolling(tas_sel).sel(time=slice(*period))
            - tas_sel.sel(time=slice(*tas_baseline_period)).mean()
        )
```
This would give a time series of warming levels. If you want a single warming level, then you would need to give a period with a single year, e.g. ("1990", "1990") (or change the input and accept a single `str`).

Reply:
Currently, this gives a single result. Taking an example from the tests:
`xs.get_warming_level_from_period("CMIP6_CanESM5_ssp585_r1i1p1f1",  ["2013", "2032"]) = 2.005`. It's the difference between the mean temperatures for 1850-1900 and 2032-2013.

For this function, since the user already provides a period, I didn't want to impose window lengths.